src/sentry/utils/http.py

```diff
@@ -133,7 +133,7 @@ def is_valid_origin(origin, project=None, allowed=None):
             parsed_hostname = parsed.hostname
 
     if parsed.port:
-        domain_matches = (
+        domain_matches: tuple[str, ...] = (
```

Comment:
How is this type understood? We have a tuple of at least one string?

If I wanted a tuple of at least two strings would I do `tuple[str, str, ...]`?

Reply:
I go over that in https://youtu.be/yLwvOwTO068 -- this is a variable-length tuple of strings

to represent "at least two strings" you need new functionality:

```python
# using compatible syntax
x: tuple[tuple[str], Unpack[tuple[str, ...]]]
# using the 3.11+ syntax
x: tuple[str, *tuple[str, ...]]
```